pow/randomx/src/lib.rs

```diff
@@ -99,32 +104,49 @@ unsafe impl<M: WithCacheMode> Send for Cache<M> { }
 unsafe impl<M: WithCacheMode> Sync for Cache<M> { }
 
 impl<M: WithCacheMode> Cache<M> {
-	pub fn new(key: &[u8], config: &Config) -> Self {
+	pub fn new(config: &Config) -> Result<Self, Error> {
```

Comment:
Change this `new` function to still take `key`, and the below function to `reinit`.

We must ensure `Cache` is usable in its entire lifetime. If someone called `new` but didn't call `init`, then all other usage of this type would fail. This is considered unsafe in Rust.

Reply:
>Change this new function to still take key, and the below function to reinit.
👌 
>This is considered unsafe in Rust.
Good to know! Thanks